backend/sqlite/sqlite.go

```diff
@@ -441,40 +444,75 @@ func (be *sqliteBackend) createOrchestrationInstanceInternal(ctx context.Context
 		return errors.New("HistoryEvent must be an ExecutionStartedEvent")
 	}
 
-	// TODO: Support for re-using orchestration instance IDs
-	res, err := tx.ExecContext(
+	var instanceExists bool
+	instanceRes, existErr := tx.QueryContext(
 		ctx,
-		`INSERT OR IGNORE INTO [Instances] (
-			[Name],
-			[Version],
-			[InstanceID],
-			[ExecutionID],
-			[Input],
-			[RuntimeStatus],
-			[CreatedTime]
-		) VALUES (?, ?, ?, ?, ?, ?, ?)`,
-		startEvent.Name,
-		startEvent.Version.GetValue(),
+		`SELECT * FROM [Instances] WHERE [InstanceID] = ?`, 
 		startEvent.OrchestrationInstance.InstanceId,
-		startEvent.OrchestrationInstance.ExecutionId.GetValue(),
-		startEvent.Input.GetValue(),
-		"PENDING",
-		e.Timestamp.AsTime(),
 	)
-	if err != nil {
-		return fmt.Errorf("failed to insert into [Instances] table: %w", err)
+	if existErr != nil {
+		return fmt.Errorf("failed to query from [Instances] table: %w", existErr)
 	}
 
-	rows, err := res.RowsAffected()
-	if err != nil {
-		return fmt.Errorf("failed to count the rows affected: %w", err)
-	}
+	if instanceRes == nil {
+		instanceExists = false
+	} else {
+		instanceExists = true
+	}
+
+	if !instanceExists || startEvent.InstanceExistOption == protos.InstanceExistOption_Throw_IF_EXIST {
+		// TODO: Support for re-using orchestration instance IDs
+		res, err := tx.ExecContext(
+			ctx,
+			`INSERT OR IGNORE INTO [Instances] (
+				[Name],
+				[Version],
+				[InstanceID],
+				[ExecutionID],
+				[Input],
+				[RuntimeStatus],
+				[CreatedTime]
+			) VALUES (?, ?, ?, ?, ?, ?, ?)`,
+			startEvent.Name,
+			startEvent.Version.GetValue(),
+			startEvent.OrchestrationInstance.InstanceId,
+			startEvent.OrchestrationInstance.ExecutionId.GetValue(),
+			startEvent.Input.GetValue(),
+			"PENDING",
+			e.Timestamp.AsTime(),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to insert into [Instances] table: %w", err)
+		}
 
-	if rows <= 0 {
-		return backend.ErrDuplicateEvent
-	}
+		rows, err := res.RowsAffected()
+		if err != nil {
+			return fmt.Errorf("failed to count the rows affected: %w", err)
+		}
+
+		if rows <= 0 {
+			return backend.ErrDuplicateEvent
+		}
 
-	*instanceID = startEvent.OrchestrationInstance.InstanceId
+		*instanceID = startEvent.OrchestrationInstance.InstanceId
+	} else if startEvent.InstanceExistOption == protos.InstanceExistOption_SKIP_IF_EXIST {
+		be.logger.Debugf(
```

Comment:
Instead of logging here, I suggest we log in the calling code. That way, we have common logging behavior that works for all backend types. I want to minimize the amount of duplication we have to do across all backends (sqlite, actors, postgres, custom, etc.)

Reply:
One question about the `Backend` interface, if I want to update it will it be a break change for Dapr workflow, I haven't really look into how Dapr is using durabletask-go, I assume it implements the `Backend` interface but using actor storage. So is it ok to update this interface?

Reply:
I don't think my suggestion requires a breaking change, but it's okay for us to make breaking changes still since `durabletask-go` is not consider "stable". Yes, we'll need to update the Dapr implementation, but that's fine - we've done it many times before. It doesn't impact customers.